contracts/src/token/erc1155/mod.rs

```diff
@@ -0,0 +1,1975 @@
+            .collect();
+
+        Ok(balances)
+    }
+
+    fn set_approval_for_all(
+        &mut self,
+        operator: Address,
+        approved: bool,
+    ) -> Result<(), Self::Error> {
+        self._set_approval_for_all(msg::sender(), operator, approved)
+    }
+
+    fn is_approved_for_all(&self, account: Address, operator: Address) -> bool {
+        self._operator_approvals.get(account).get(operator)
+    }
+
+    fn safe_transfer_from(
+        &mut self,
+        from: Address,
+        to: Address,
+        id: U256,
+        value: U256,
+        data: Bytes,
+    ) -> Result<(), Self::Error> {
+        self.authorize_transfer(from)?;
+        self.do_safe_transfer_from(from, to, vec![id], vec![value], &data)
+    }
+
+    fn safe_batch_transfer_from(
+        &mut self,
+        from: Address,
+        to: Address,
+        ids: Vec<U256>,
+        values: Vec<U256>,
+        data: Bytes,
+    ) -> Result<(), Self::Error> {
+        self.authorize_transfer(from)?;
+        self.do_safe_transfer_from(from, to, ids, values, &data)
+    }
+}
+
+impl IErc165 for Erc1155 {
+    fn supports_interface(interface_id: FixedBytes<4>) -> bool {
+        <Self as IErc1155>::INTERFACE_ID == u32::from_be_bytes(*interface_id)
+            || Erc165::supports_interface(interface_id)
+    }
+}
+
+impl Erc1155 {
+    /// Transfers a `value` amount of tokens of type `ids` from `from` to
+    /// `to`. Will mint (or burn) if `from` (or `to`) is the `Address::ZERO`.
+    ///
+    /// # Arguments
+    ///
+    /// * `&mut self` - Write access to the contract's state.
+    /// * `from` - Account to transfer tokens from.
+    /// * `to` - Account of the recipient.
+    /// * `ids` - Array of all tokens ids.
+    /// * `values` - Array of all amount of tokens to be transferred.
+    ///
+    /// # Errors
+    ///
+    /// If length of `ids` is not equal to length of `values`, then the
+    /// error [`Error::InvalidArrayLength`] is returned.
+    /// If `value` is greater than the balance of the `from` account,
+    /// then the error [`Error::InsufficientBalance`] is returned.
+    ///
+    /// NOTE: The ERC-1155 acceptance check is not performed in this function.
+    /// See [`Self::_update_with_acceptance_check`] instead.
+    ///
+    /// # Events
+    ///
+    /// Emits a [`TransferSingle`] event if the arrays contain one element, and
+    /// [`TransferBatch`] otherwise.
+    ///
+    /// # Panics
+    ///
+    /// If updated balance exceeds `U256::MAX`, may happen during `mint`
+    /// operation.
+    fn _update(
+        &mut self,
+        from: Address,
+        to: Address,
+        ids: Vec<U256>,
+        values: Vec<U256>,
+    ) -> Result<(), Error> {
+        Self::require_equal_arrays(&ids, &values)?;
+
+        let operator = msg::sender();
+
+        ids.iter().zip(values.iter()).try_for_each(|(&token_id, &value)| {
+            self.do_update(from, to, token_id, value)
+        })?;
```

Comment:
NIt: may be use a for loop here?
```suggestion
        for (&token_id, &value) in ids.iter().zip(values.iter()) {
            self.do_update(from, to, token_id, value)?;
        }
```

Reply:
That may be more readable 👍